src/libraries/Native/build-native.sh

```diff
@@ -78,6 +78,14 @@ fi
 if [[ "$__TargetOS" == OSX ]]; then
     # set default OSX deployment target
     __CMakeArgs="-DCMAKE_OSX_DEPLOYMENT_TARGET=10.13 $__CMakeArgs"
+    if [[ "$__BuildArch" == x64 ]]; then
+        __CMakeArgs="-DCMAKE_OSX_ARCHITECTURES=\"x86_64\" $__CMakeArgs"
+    elif [[ "$__BuildArch" == arm64 ]]; then
+        __CMakeArgs="-DCMAKE_OSX_ARCHITECTURES=\"arm64\" $__CMakeArgs"
+    else
+        echo "Error: Unknown OSX architecture $__BuildArch."
+        exit 1
+    fi
```

Comment:
Without this code libraries was build Host=x64 Target=arm64 native libraries.

I suspect we need something like this in `src/coreclr/build-runtime.sh` as well.  For some reason last time I built 
`-subset clr.runtime`, it ended up failing because it was building this incorect combo.

It is possible this should move to a common script too...  But it is very script specific here.